ocs_ci/deployment/hosted_cluster.py

```diff
@@ -112,51 +119,39 @@ def do_deploy(self):
         # stage 5 verify ODF client is installed on all hosted clusters
         odf_installed = []
         for cluster_name in cluster_names:
-
-            if not self.config_has_hosted_odf_image(cluster_name):
+            if self.config_has_hosted_odf_image(cluster_name):
                 logger.info(
-                    f"Hosted ODF image not set for cluster '{cluster_name}', skipping ODF validation"
+                    f"Validate ODF client operator installed on hosted OCP cluster '{cluster_name}'"
                 )
-                continue
-
-            logger.info(
-                f"Validate ODF client operator installed on hosted OCP cluster '{cluster_name}'"
-            )
-            hosted_odf = HostedODF(cluster_name)
-
-            if not hosted_odf.odf_client_installed():
-                # delete catalogsources help to finish install cluster if nodes have not enough mem
-                # see oc describe pod ocs-client-operator-controller-manager-<suffix> -n openshift-storage-client
-                # when the problem was hit
-                hosted_odf.exec_oc_cmd(
-                    "delete catalogsource --all -n openshift-marketplace"
-                )
-                logger.info("wait 30 sec and create catalogsource again")
-                time.sleep(30)
-                hosted_odf.create_catalog_source()
-            odf_installed.append(hosted_odf.odf_client_installed())
+                hosted_odf = HostedODF(cluster_name)
+                if not hosted_odf.odf_client_installed():
+                    hosted_odf.exec_oc_cmd(
+                        "delete catalogsource --all -n openshift-marketplace"
+                    )
+                    logger.info("wait 30 sec and create catalogsource again")
+                    time.sleep(30)
```

Comment:
can we instead of sleep sample the existence of the catalogsource to validate it has been deleted?

Reply:
* I strongly agree that static sleeps are bad but I've made an exception here.
* This 30-second timeout was added as a workaround for a very rare issue. We will not hit it 99% of the times 
* we don't wait for only catalog source to be deleted, but wait for leftovers associated with this catalog source to be cleaned up too (I don't know exactly the full list, images, pods, secrets..). 
* we remove no one catalogsource but all non-default catalogsources from namespace